---
float64[] configuration
bool valid